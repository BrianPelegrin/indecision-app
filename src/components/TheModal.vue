<script setup lang="ts">
interface IProps{
    modelValue:boolean;
    persistent?:boolean;
}
interface IEmits{
    (event:'clickBackdrop'):void
}
defineEmits<IEmits>()
defineProps<IProps>()
</script>

<template>
    <div 
        v-if="modelValue" 
        @click="()=>{
            if(!persistent){
                $emit('clickBackdrop')
            }
        }" 
        class="top-0 fixed  sticky-0  flex justify-center items-center h-full w-full"
        style="background-color: rgb(243 244 246 / 0.7);"
    >
        <div class="flex justify-center items-center" @click.stop="">
            <slot />
        </div>
    </div>    
</template>

<style scoped>

</style>
