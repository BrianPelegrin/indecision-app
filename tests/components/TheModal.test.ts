import {test, describe, expect} from "vitest"
import { mount } from "@vue/test-utils"
import TheModal from "@/components/TheModal.vue"

describe('<TheModal />',()=>{

    test('Should match the modal snapshot',()=>{
        const wrapper = mount(TheModal,{
            props:{
                modelValue:true
            }
        })

        expect(wrapper).toMatchSnapshot()
        
        
    })

})
