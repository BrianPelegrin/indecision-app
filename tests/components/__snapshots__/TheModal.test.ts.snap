// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TheModal /> > Should match the modal snapshot 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="top-0 fixed sticky-0 flex justify-center items-center h-full w-full"
      >
        <div
          class="flex justify-center items-center"
        >
          
          
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": Dep {
            "activeLink": undefined,
            "computed": undefined,
            "subs": undefined,
            "subsHead": undefined,
            "version": 1,
          },
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "cleanup": undefined,
        "deps": {
          "dep": Dep {
            "activeLink": undefined,
            "computed": undefined,
            "subs": [Circular],
            "subsHead": [Circular],
            "version": 0,
          },
          "nextDep": {
            "dep": Dep {
              "activeLink": undefined,
              "computed": undefined,
              "subs": [Circular],
              "subsHead": [Circular],
              "version": 0,
            },
            "nextDep": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": undefined,
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": [Circular],
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": [Circular],
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "nextSub": undefined,
            "prevActiveLink": undefined,
            "prevDep": [Circular],
            "prevSub": undefined,
            "sub": [Circular],
            "version": 0,
          },
          "nextSub": undefined,
          "prevActiveLink": undefined,
          "prevDep": undefined,
          "prevSub": undefined,
          "sub": [Circular],
          "version": 0,
        },
        "depsTail": {
          "dep": Dep {
            "activeLink": undefined,
            "computed": undefined,
            "subs": [Circular],
            "subsHead": [Circular],
            "version": 0,
          },
          "nextDep": undefined,
          "nextSub": undefined,
          "prevActiveLink": undefined,
          "prevDep": {
            "dep": Dep {
              "activeLink": undefined,
              "computed": undefined,
              "subs": [Circular],
              "subsHead": [Circular],
              "version": 0,
            },
            "nextDep": [Circular],
            "nextSub": undefined,
            "prevActiveLink": undefined,
            "prevDep": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": [Circular],
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": [Circular],
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": undefined,
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "prevSub": undefined,
            "sub": [Circular],
            "version": 0,
          },
          "prevSub": undefined,
          "sub": [Circular],
          "version": 0,
        },
        "flags": 37,
        "fn": [Function],
        "nextEffect": undefined,
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "ids": [
        "",
        0,
        0,
      ],
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "job": [Function],
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "_isPaused": false,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "cleanup": undefined,
            "deps": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": {
                  "dep": Dep {
                    "activeLink": undefined,
                    "computed": undefined,
                    "subs": [Circular],
                    "subsHead": [Circular],
                    "version": 0,
                  },
                  "nextDep": {
                    "dep": Dep {
                      "activeLink": undefined,
                      "computed": undefined,
                      "subs": [Circular],
                      "subsHead": [Circular],
                      "version": 0,
                    },
                    "nextDep": undefined,
                    "nextSub": undefined,
                    "prevActiveLink": undefined,
                    "prevDep": [Circular],
                    "prevSub": undefined,
                    "sub": [Circular],
                    "version": 0,
                  },
                  "nextSub": undefined,
                  "prevActiveLink": undefined,
                  "prevDep": [Circular],
                  "prevSub": undefined,
                  "sub": [Circular],
                  "version": 0,
                },
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": [Circular],
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": undefined,
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "depsTail": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": undefined,
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": [Circular],
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": {
                  "dep": Dep {
                    "activeLink": undefined,
                    "computed": undefined,
                    "subs": [Circular],
                    "subsHead": [Circular],
                    "version": 0,
                  },
                  "nextDep": [Circular],
                  "nextSub": undefined,
                  "prevActiveLink": undefined,
                  "prevDep": {
                    "dep": Dep {
                      "activeLink": undefined,
                      "computed": undefined,
                      "subs": [Circular],
                      "subsHead": [Circular],
                      "version": 0,
                    },
                    "nextDep": [Circular],
                    "nextSub": undefined,
                    "prevActiveLink": undefined,
                    "prevDep": undefined,
                    "prevSub": undefined,
                    "sub": [Circular],
                    "version": 0,
                  },
                  "prevSub": undefined,
                  "sub": [Circular],
                  "version": 0,
                },
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "flags": 37,
            "fn": [Function],
            "nextEffect": undefined,
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "modelValue": true,
            "persistent": false,
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "cleanup": undefined,
            "deps": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": undefined,
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": [Circular],
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": undefined,
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "depsTail": {
              "dep": Dep {
                "activeLink": undefined,
                "computed": undefined,
                "subs": [Circular],
                "subsHead": [Circular],
                "version": 0,
              },
              "nextDep": undefined,
              "nextSub": undefined,
              "prevActiveLink": undefined,
              "prevDep": {
                "dep": Dep {
                  "activeLink": undefined,
                  "computed": undefined,
                  "subs": [Circular],
                  "subsHead": [Circular],
                  "version": 0,
                },
                "nextDep": [Circular],
                "nextSub": undefined,
                "prevActiveLink": undefined,
                "prevDep": undefined,
                "prevSub": undefined,
                "sub": [Circular],
                "version": 0,
              },
              "prevSub": undefined,
              "sub": [Circular],
              "version": 0,
            },
            "flags": 37,
            "fn": [Function],
            "nextEffect": undefined,
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": {
            "clickBackdrop": null,
          },
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "ids": [
            "",
            0,
            0,
          ],
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "job": [Function],
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "modelValue": true,
            "persistent": false,
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "modelValue": {
                "0": true,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "persistent": {
                "0": true,
                "1": true,
                "required": false,
                "type": [Function],
              },
            },
            [
              "modelValue",
              "persistent",
            ],
          ],
          "provides": {},
          "proxy": {
            "modelValue": true,
            "persistent": false,
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "_isPaused": false,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "cleanup": undefined,
                "deps": {
                  "dep": Dep {
                    "activeLink": undefined,
                    "computed": undefined,
                    "subs": [Circular],
                    "subsHead": [Circular],
                    "version": 0,
                  },
                  "nextDep": {
                    "dep": Dep {
                      "activeLink": undefined,
                      "computed": undefined,
                      "subs": [Circular],
                      "subsHead": [Circular],
                      "version": 0,
                    },
                    "nextDep": undefined,
                    "nextSub": undefined,
                    "prevActiveLink": undefined,
                    "prevDep": [Circular],
                    "prevSub": undefined,
                    "sub": [Circular],
                    "version": 0,
                  },
                  "nextSub": undefined,
                  "prevActiveLink": undefined,
                  "prevDep": undefined,
                  "prevSub": undefined,
                  "sub": [Circular],
                  "version": 0,
                },
                "depsTail": {
                  "dep": Dep {
                    "activeLink": undefined,
                    "computed": undefined,
                    "subs": [Circular],
                    "subsHead": [Circular],
                    "version": 0,
                  },
                  "nextDep": undefined,
                  "nextSub": undefined,
                  "prevActiveLink": undefined,
                  "prevDep": {
                    "dep": Dep {
                      "activeLink": undefined,
                      "computed": undefined,
                      "subs": [Circular],
                      "subsHead": [Circular],
                      "version": 0,
                    },
                    "nextDep": [Circular],
                    "nextSub": undefined,
                    "prevActiveLink": undefined,
                    "prevDep": undefined,
                    "prevSub": undefined,
                    "sub": [Circular],
                    "version": 0,
                  },
                  "prevSub": undefined,
                  "sub": [Circular],
                  "version": 0,
                },
                "flags": 37,
                "fn": [Function],
                "nextEffect": undefined,
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [],
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": ,
                    "key": "_default",
                    "patchFlag": -2,
                    "props": {
                      "key": "_default",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "targetStart": null,
                    "transition": null,
                    "type": Symbol(v-fgt),
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="flex justify-center items-center"
                >
                  
                  
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "flex justify-center items-center",
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": Symbol(v-fgt),
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="top-0 fixed sticky-0 flex justify-center items-center h-full w-full"
            >
              <div
                class="flex justify-center items-center"
              >
                
                
              </div>
            </div>,
            "key": 0,
            "patchFlag": 0,
            "props": {
              "class": "top-0 fixed sticky-0 flex justify-center items-center h-full w-full",
              "key": 0,
              "onClick": [Function],
              "style": {
                "background-color": "rgb(243 244 246 / 0.7)",
              },
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "targetStart": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/home/brianpelegrin/Documents/vue-projects/indecision-app/src/components/TheModal.vue",
            "__name": "TheModal",
            "components": {},
            "emits": [
              "clickBackdrop",
            ],
            "props": {
              "modelValue": {
                "required": true,
                "type": [Function],
              },
              "persistent": {
                "required": false,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="top-0 fixed sticky-0 flex justify-center items-center h-full w-full"
        >
          <div
            class="flex justify-center items-center"
          >
            
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "modelValue": true,
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "__file": "/home/brianpelegrin/Documents/vue-projects/indecision-app/src/components/TheModal.vue",
          "__name": "TheModal",
          "components": {},
          "emits": [
            "clickBackdrop",
          ],
          "props": {
            "modelValue": {
              "required": true,
              "type": [Function],
            },
            "persistent": {
              "required": false,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="top-0 fixed sticky-0 flex justify-center items-center h-full w-full"
        >
          <div
            class="flex justify-center items-center"
          >
            
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "onUnmount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.5.3",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="top-0 fixed sticky-0 flex justify-center items-center h-full w-full"
  >
    <div
      class="flex justify-center items-center"
    >
      
      
    </div>
  </div>,
}
`;
